server/graphql/types/src/types/program/contact_trace.rs

```diff
@@ -0,0 +1,286 @@
+        let Some(ref contact_patient_id) = self.trace_row().contact_patient_id else {
+          return Ok(None)
+      };
+        let loader = ctx.get_loader::<DataLoader<PatientLoader>>();
+
+        let result = loader
+            .load_one(contact_patient_id.clone())
+            .await?
+            .map(|patient| PatientNode {
+                store_id: self.store_id.clone(),
+                allowed_ctx: self.allowed_ctx.clone(),
+                patient,
+            })
+            .ok_or(Error::new("Contact without patient"))?;
+
+        Ok(Some(result))
+    }
+
+    /// Returns the matching program enrolment for the root patient of this contact trace
+    pub async fn program_enrolment(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Option<ProgramEnrolmentNode>> {
+        let loader = ctx.get_loader::<DataLoader<ProgramEnrolmentLoader>>();
+
+        let result = loader
+            .load_one(ProgramEnrolmentLoaderInput::new(
+                &self.trace_row().patient_id,
+                &self.trace_row().program_id,
+                self.allowed_ctx.clone(),
+            ))
+            .await?
+            .map(|program_enrolment| ProgramEnrolmentNode {
+                store_id: self.store_id.clone(),
+                program_enrolment,
+                allowed_ctx: self.allowed_ctx.clone(),
+            })
+            .ok_or(Error::new(format!(
+                "Failed to load program enrolment: {}",
+                self.trace_row().program_id
+            )))?;
+
+        Ok(Some(result))
+    }
+
+    pub async fn datetime(&self) -> DateTime<Utc> {
+        DateTime::<Utc>::from_utc(self.trace_row().datetime, Utc)
+    }
+
+    /// The encounter document
+    pub async fn document(&self, ctx: &Context<'_>) -> Result<DocumentNode> {
+        let loader = ctx.get_loader::<DataLoader<DocumentByIdLoader>>();
+
+        let result = loader
+            .load_one(self.trace_row().document_id.clone())
+            .await?
+            .map(|document| DocumentNode {
+                allowed_ctx: self.allowed_ctx.clone(),
+                document,
+            })
+            .ok_or(Error::new("Missing contact trace document"))?;
+
+        Ok(result)
+    }
+
+    pub async fn first_name(&self) -> Option<String> {
+        self.trace_row().first_name.clone()
+    }
+
+    pub async fn last_name(&self) -> Option<String> {
+        self.trace_row().last_name.clone()
+    }
+
+    pub async fn gender(&self) -> Option<GenderType> {
+        self.trace_row()
+            .gender
+            .as_ref()
+            .map(GenderType::from_domain)
+    }
+
+    pub async fn date_of_birth(&self) -> Option<NaiveDate> {
+        self.trace_row().date_of_birth.clone()
+    }
+
+    pub async fn age(&self) -> Option<i64> {
+        self.trace_row().date_of_birth.clone().map(|dob| {
+            let diff = Local::now().naive_utc().date().signed_duration_since(dob);
+            diff.num_days() / 365
```

Comment:
Could use 
https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.years_since
?
I guess we're assuming everyone is > 1 years old?

Reply:
Yes good idea, will do next time since this seems to be working...

> I guess we're assuming everyone is > 1 years old?

What do you mean? new born is 0 year isn't it?

Reply:
Yeah I guess so.
For this purpose I think it's fine.
If it was a paediatrician focus we might want to handle things like 9months vs 3months where the difference could be significant.